test/helpers/view_components/notification_icon_test.rb

```diff
@@ -0,0 +1,11 @@
+require_relative "./view_component_test_case"
+
+class NotificationIconTest < ViewComponentTestCase
+  test "#notification_icon passes the correct parameters" do
+    user = create(:user)
+    create(:notification, user: user, read_at: nil)
+
+    assert_component_equal ViewComponents::NotificationIcon.new(user).to_s,
+                           { id: "notification-icon", props: { count: 1 } }
+  end
+end
```

Comment:
Can you wizz these up for the other view components pls?

Reply:
I've added tests for the other components. There is a bug in the test verification though. It has to do with how the `assert_component_equal` method is implemented:

```ruby
def assert_component_equal(component, params)
    assert_dom_equal(
      %(<div data-react-#{params[:id]}="true" data-react-data=#{params[:props].to_json} />),
      component
    )
  end
```

The new tests have a `props` argument where there are strings in the values that are quoted in the output. I've briefly looked into this, but figured it is probably much more efficient for one of you to fix this (or point me to the fix) :)

```
Expected: <div data-react-maintaining-iterations-summary-table="true" data-react-data={"solution_id":1,"iterations":[{"id":1,"testsStatus":"pending"},{"id":2,"testsStatus":"passed"}]} />
Actual: <div data-react-example-iterations-summary-table="true" data-react-data="{&quot;solution_id&quot;:90,&quot;iterations&quot;:[{&quot;id&quot;:53,&quot;testsStatus&quot;:&quot;pending&quot;},{&quot;id&quot;:54,&quot;testsStatus&quot;:&quot;passed&quot;}]}" />
```

Reply:
@kntsoriano One for you :) If the `j` method (alias for `escape_javascript`) is available here that might work?